### DISCUSSION TITLE
questions about Input data types as of August 2024

### DISCUSSION THREAD

**Discussion by flight505** [microsoft/graphrag#960] [Q&A]
A few trivial questions, I just started using MS GraphRAG. I have some healthcare data including patients associated with almost 200 attributes. It was found that using JSON was preferred as a data type because it retained more metadata that could be utilized by KG (not ms grapgrag). However, I didn't see it as a direct input option in the documentation for Microsoft GraphRAG system. Using CSV with several title, timestamps, and source fields seems to be a bit tricky. When the text corpus is segmented into analyzable units, extracting entities and relationships gives me very different results compared to for example neo4j

Here are my questions:

1) Can I adjust the entity extraction and relationship definitions based on specific requirements, or do I need to customize it from the beginning?
2) All the PDFs I use were converted to .md as this preserves the data table structures from the documents. Directly using .md for input seems possible but is not mentioned in the documentation. Does anyone know how .md is treated differently in the DataShaper workflow, or is there something to be aware of?
3) I would imagine that most file types supported by OpenAI API can be ingested?
4) Last question, I am looking for good sources for in-depth use cases of Microsoft GraphRAG, not the same minimalistic getting started guides that hundreds of "content creators" have duplicated.

**Comment by xuezhizeng** [comment]
This is really good question. I am looking for the answers as well.
